design-documents/graph-ql/coverage/b2b/negotiableQuotes.graphqls

```diff
@@ -190,6 +205,7 @@ type NegotiableQuote {
     created_at: String @doc(description: "Timestamp indicating when the negotiable quote was created.")
     updated_at: String @doc(description: "Timestamp indicating when the negotiable quote was updated.")
     status: NegotiableQuoteStatus!
+    shipping_addresses: [ShippingCartAddress!]!
```

Comment:
I'd do the same thing as above with an interface and reorder the naming to `QuoteShippingAddress` on the new types (rather than `ShippingQuoteAddress`).

Reply:
But leave the name `ShippingCartAddress` as is, right?

Reply:
Also, why is `ShippingCartAddress` required here?

Reply:
The outer exclamation point indicates that the value cannot be null. In this case we expect, at the very least, an empty array. 

The inner exclamation point indicates that if the array is not empty, every element in the array must be of type `ShippingCartAddress`.

Reply:
We don't have the inner exclamation in the QuoteGraphQl schema's field, though.

Reply:
> I'd do the same thing as above with an interface and reorder the naming to `QuoteShippingAddress` on the new types (rather than `ShippingQuoteAddress`).

We should actually think on this some more.  Having a different name for this one would mean anything that extends `ShippingCartAddress` would not be applied to `QuoteShippingAddress` even though they implement the same interface.



> But leave the name `ShippingCartAddress` as is, right?

Yes, we have to leave the existing names in place for BiC.

Reply:
> We don't have the inner exclamation in the QuoteGraphQl schema's field, though.

I'm aware of this. My logic was that the NegotiableQuote schema already deviated from the Cart (specifically the items). It seemed appropriate to structure the NegotiableQuote shipping_addresses property as it _should_ be. It's also easy to go from more restrictive to less restrictive.

However, I'm not steadfast on this. For the purpose of unifying these objects under a common interface, it might be better to define them as closely as possible.